src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -51,6 +51,55 @@ namespace winrt::TerminalApp::implementation
         InitializeComponent();
     }
 
+    // Function Description:
+    // - Recursively check our commands to see if there's a keybinding for
+    //   exactly their action. If there is, label that command with the text
+    //   corresponding to that key chord.
+    // - Will recurse into nested commands as well.
+    // Arguments:
+    // - settings: The settings who's keybindings we should use to look up the key chords from
+    // - commands: The list fo commands to label.
+    // Return Value:
+    // - <none>
+    static void _recursiveUpdateCommandKeybindingLabels(std::shared_ptr<::TerminalApp::CascadiaSettings> settings,
+                                                        Windows::Foundation::Collections::IVector<TerminalApp::Command> const& commands)
+    {
+        for (auto command : commands)
+        {
+            // If there's a keybinding that's bound to exactly this command,
+            // then get the string for that keychord and display it as a
+            // part of the command in the UI. Each Command's KeyChordText is
+            // unset by default, so we don't need to worry about clearing it
+            // if there isn't a key associated with it.
+            auto keyChord{ settings->GetKeybindings().GetKeyBindingForActionWithArgs(command.Action()) };
+
+            if (keyChord)
+            {
+                command.KeyChordText(KeyChordSerialization::ToString(keyChord));
+            }
+
+            _recursiveUpdateCommandKeybindingLabels(settings, command.NestedCommands());
+        }
+    }
+
+    static void _recursiveUpdateCommandIcons(Windows::Foundation::Collections::IVector<TerminalApp::Command> const& commands)
```

Comment:
I'm curious, why is this function is separate from `_recursiveUpdateCommandKeybindingLabels`?

Reply:
I don't really know. I thought maybe there'd be some other logic I'd have to do for the icons and wanted to keep it separate? I'll merge 'em